File: Examples/Python/usb_connection.py
import time

import connection
import helpers
import ximu3

if helpers.yes_or_no("Search for connections?"):
    devices = ximu3.PortScanner.scan_filter(ximu3.PORT_TYPE_USB)

    time.sleep(1)  # wait for OS to release port

    if not devices:
        raise Exception("No USB connections available")

    print(f"Found {devices[0]}")

    config = devices[0].connection_config

    connection.run(config)
else:
    config = ximu3.UsbConnectionConfig("COM1")  # replace with actual connection config

    connection.run(config)

File: Examples/CSharp/x-IMU3-Examples/Examples/UsbConnection.cs
namespace Ximu3Examples
{
    class UsbConnection : Connection
    {
        public UsbConnection()
        {
            if (Helpers.YesOrNo("Connect automatically?"))
            {
                var devices = Ximu3.PortScanner.ScanFilter(Ximu3.CApi.XIMU3_PortType.XIMU3_PortTypeUsb);

                System.Threading.Thread.Sleep(1000); // wait for OS to release port

                if (devices.Length == 0)
                {
                    Console.WriteLine("No USB connections available");
                    return;
                }

                Console.WriteLine("Found " + devices[0].ToString());

                var config = Ximu3.ConnectionConfig.From(devices[0])!;

                Run(config);
            }
            else
            {
                var config = new Ximu3.UsbConnectionConfig("COM1"); // replace with actual connection config

                Run(config);
            }
        }
    }
}
